src/metrics/mongoose_metrics_hooks.erl

```diff
@@ -139,25 +127,48 @@ xmpp_stanza_dropped(Acc, _, _, _) ->
     mongoose_metrics:update(HostType, xmppStanzaDropped, 1),
     Acc.
 
--spec xmpp_send_element(Acc :: mongoose_acc:t(), Server :: jid:server()) -> mongoose_acc:t().
-xmpp_send_element(Acc, _El) ->
+-spec xmpp_send_element(Acc :: mongoose_acc:t(), El :: exml:element()) -> mongoose_acc:t().
+xmpp_send_element(Acc, #xmlel{name = Name} = El)
+  when Name == <<"iq">>; Name == <<"message">>; Name == <<"presence">> ->
     HostType = mongoose_acc:host_type(Acc),
     mongoose_metrics:update(HostType, xmppStanzaCount, 1),
-    case mongoose_acc:stanza_type(Acc) of
+    case exml_query:attr(El, <<"type">>) of
         <<"error">> ->
             mongoose_metrics:update(HostType, xmppErrorTotal, 1),
-            case mongoose_acc:stanza_name(Acc) of
-                <<"iq">> ->
-                    mongoose_metrics:update(HostType, xmppErrorIq, 1);
-                <<"message">> ->
-                    mongoose_metrics:update(HostType, xmppErrorMessage, 1);
-                <<"presence">> ->
-                    mongoose_metrics:update(HostType, xmppErrorPresence, 1)
-            end;
-        _ -> ok
+            xmpp_send_element_error(HostType, El);
+        _ ->
+            mongoose_metrics:update(HostType, xmppStanzaReceived, 1),
+            xmpp_send_element_type(HostType, El)
     end,
+    Acc;
+xmpp_send_element(Acc, _El) ->
     Acc.
 
+-spec xmpp_send_element_error(HostType :: mongooseim:host_type(),
+                              Packet :: exml:element()) -> ok | {error, not_found}.
+xmpp_send_element_error(HostType, #xmlel{name = <<"message">>}) ->
+    mongoose_metrics:update(HostType, xmppErrorMessage, 1);
+xmpp_send_element_error(HostType, #xmlel{name = <<"iq">>}) ->
+    mongoose_metrics:update(HostType, xmppErrorIq, 1);
+xmpp_send_element_error(HostType, #xmlel{name = <<"presence">>}) ->
+    mongoose_metrics:update(HostType, xmppErrorPresence, 1).
```

Comment:
Btw, have you noticed that there are packets that are not stanzas, that is, they're not of type message/presence/iq, and therefore this pattern-matching will fail and throw a dirty log? 😄 

Ideally we should use here the other new hooks, and have `user_send_packet` for counting packets, and `user_send_(message|presence|iq)` for exactly counting the three standard stanzas. But that's more upcoming conflicts with the hooks rework. Though just mechanical conflicts, maybe we can afford it already, what do you think? 😃 

PS: same thing will apply to the other functions matching on the stanza kind.